// angular-captcha/src/captcha.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var captcha_service_1 = require("./captcha.service");
var captcha_helper_service_1 = require("./captcha-helper.service");
var CaptchaComponent = /** @class */ (function () {
    function CaptchaComponent(elementRef, captchaService, captchaHelper) {
        this.elementRef = elementRef;
        this.captchaService = captchaService;
        this.captchaHelper = captchaHelper;
    }
    Object.defineProperty(CaptchaComponent.prototype, "captchaId", {
        // The current captcha id, which will be used for validation purpose.
        get: function () {
            return this.captchaService.botdetectInstance.captchaId;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CaptchaComponent.prototype, "captchaCode", {
        // The typed captcha code value.
        get: function () {
            return this.captchaService.botdetectInstance.userInput.value;
        },
        enumerable: true,
        configurable: true
    });
    // Display captcha html markup on component initialization.
    CaptchaComponent.prototype.ngOnInit = function () {
        // if styleName is not specified, the styleName will be 'defaultCaptcha'
        if (!this.styleName) {
            this.styleName = 'defaultCaptcha';
        }
        // set captcha style name to CaptchaService for creating BotDetect object
        this.captchaService.styleName = this.styleName;
        // display captcha html markup on view
        this.displayHtml();
    };
    // Display captcha html markup in the <botdetect-captcha> tag.
    CaptchaComponent.prototype.displayHtml = function () {
        var _this = this;
        this.captchaService.getHtml()
            .subscribe(function (captchaHtml) {
            // display captcha html markup
            _this.elementRef.nativeElement.innerHTML = captchaHtml.replace(/<script.*<\/script>/g, '');
            // load botdetect scripts
            _this.loadScriptIncludes();
        }, function (error) {
            throw new Error(error);
        });
    };
    // Reload a new captcha image.
    CaptchaComponent.prototype.reloadImage = function () {
        this.captchaService.botdetectInstance.reloadImage();
    };
    // Validate captcha on client-side and execute user callback function on ajax success
    CaptchaComponent.prototype.validateUnsafe = function (callback) {
        var _this = this;
        var userInput = this.captchaService.botdetectInstance.userInput;
        var captchaCode = userInput.value;
        if (captchaCode.length !== 0) {
            this.captchaService.validateUnsafe(captchaCode)
                .subscribe(function (isHuman) {
                callback(isHuman);
                if (!_this.captchaHelper.useUserInputBlurValidation(userInput) && !isHuman) {
                    _this.reloadImage();
                }
            }, function (error) {
                throw new Error(error);
            });
        }
        else {
            var isHuman = false;
            callback(isHuman);
        }
    };
    // Load BotDetect scripts.
    CaptchaComponent.prototype.loadScriptIncludes = function () {
        var captchaId = this.elementRef.nativeElement.querySelector('#BDC_VCID_' + this.styleName).value;
        var scriptIncludeUrl = this.captchaService.captchaEndpoint + '?get=script-include&c=' + this.styleName + '&t=' + captchaId + '&cs=201';
        this.captchaHelper.getScript(scriptIncludeUrl);
    };
    CaptchaComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'botdetect-captcha',
                    template: ''
                },] },
    ];
    /** @nocollapse */
    CaptchaComponent.ctorParameters = function () { return [
        { type: core_1.ElementRef, },
        { type: captcha_service_1.CaptchaService, },
        { type: captcha_helper_service_1.CaptchaHelperService, },
    ]; };
    CaptchaComponent.propDecorators = {
        'styleName': [{ type: core_1.Input },],
    };
    return CaptchaComponent;
}());
exports.CaptchaComponent = CaptchaComponent;
